Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -115,19 +80,15 @@ public SeedData Cross(SeedData a, SeedData b)
         CrossFloat(ref result.Production, a.Production);
         CrossFloat(ref result.Potency, a.Potency);
 
-        // we do not transfer Sentient to another plant to avoid ghost role spam
         CrossBool(ref result.Seedless, a.Seedless);
-        CrossBool(ref result.Viable, a.Viable);
-        CrossBool(ref result.Slip, a.Slip);
         CrossBool(ref result.Ligneous, a.Ligneous);
-        CrossBool(ref result.Bioluminescent, a.Bioluminescent);
         CrossBool(ref result.TurnIntoKudzu, a.TurnIntoKudzu);
         CrossBool(ref result.CanScream, a.CanScream);
 
         CrossGasses(ref result.ExudeGasses, a.ExudeGasses);
         CrossGasses(ref result.ConsumeGasses, a.ConsumeGasses);
 
-        result.BioluminescentColor = Random(0.5f) ? a.BioluminescentColor : result.BioluminescentColor;
+        result.Mutations = result.Mutations.Where(m => Random(0.5f)).Union(a.Mutations.Where(m => Random(0.5f))).DistinctBy(m => m.Name).ToList();
```

Comment:
I'm not super familiar with LINQ and so it could be helpful to leave a comment for passers-by as to what the intent of this LINQ statement is.

In particular, since you've already selected from `result.Mutations` with 50% chance, why do you need to select 50% randomly from `a.Mutations`, instead of just unioning with `a.Mutations`?

The cross logic used to be simply "take A 50% of the time, and if not A then B".

Reply:
I can add the comment clarifying it. I'll do a long explanation here for you and figure out a way to shorten it in the file if I can.

Since Mutations is now a list of entity effects, this takes each entry on the first plant (result.Mutations) 50% of the time, then takes each entry on the 2nd plant (a.Mutations) 50% of the time, merges both of those together in the Union call, and Distinct keeps only entries with a unique name (discarding the second one in case both list had it and both rolled that 50% to add it).

I think this is the most performant way to put this line together. Alternates would be making a new result list, putting each thing from both plants Mutations list on there 50% of the time if the results list doesn't already have an entry with the same name, and making that results.Mutations. 

I could also do the Union first, and then pick all the present entries in the combined list 50% of the time and Distinct to confirm there's no duplicates, but that does more copying/deleting memory work because the list is bigger to start and removes things from it.

Reply:
Thanks for the summary. If I understand the change in behavior correctly:

### Old Behavior
For each type of mutation, 50% chance to take from plant A, otherwise take from plant B.

### New Behavior
For each type of mutation, 50% chance to take from plant A, otherwise there's a 50% of the remaining chance to take from plant B (so overall 25% chance), and then a 25% chance of losing a mutation from crossing.

Is my understanding correct, and if so, was this the intent?

Reply:
For all of the stats on SeedData, Cross() works the way it did before.  Note that the mutations that apply changes directly to stats and parameters on SeedData are set to fire and change the value, then not persist. This list is for special effects that occur to the plant and/or its produce (currently Bioluminescent, Slippery, and Sentient. This PR enables the potential for new effects to be added without checking for each one each time, and I have roughly a dozen in mind for future plans in my Hydro rework doc).

The 2 checks for the 2 lists are independent.

Control Example: P1 and P2 have no effect mutation: The resulting cross will have no effect mutations.

EX 1: P1 has Bioluminescent, P2 does not. There is a 50% chance the resulting cross has Bioluminescent.

EX 2: P1 has Bioluminescent and Slippery, P2 has Slippery. There is a 50% chance the resulting cross has Bioluminescent, and a 75% it has Slippery. (1/4 both checks are false, 1/4 both checks are true and Distinct removes the duplicate, 2/4 one is true and the other is false). This matches up with typical Mendelian genetics if we pretend for now that all plants have 1 dominant and 1 recessive gene involved in these checks.